Source/DafnyStandardLibraries/src/DafnyStdLibs/EnableNonLinearArithmetic/Collections/Seqs.dfy

```diff
@@ -73,12 +73,23 @@ module DafnyStdLibs.Collections.Seqs {
   {
   }
 
-  /* The concatenation of sequences is associative. */
+  /* The concatenation of sequences is associative (three arguments). */
   lemma LemmaConcatIsAssociative<T>(xs: seq<T>, ys: seq<T>, zs: seq<T>)
-    ensures xs + (ys + zs) == (xs + ys) + zs
+    ensures xs + (ys + zs) == xs + ys + zs == (xs + ys) + zs
   {
   }
 
+  /* The concatenation of sequences is associative (four arguments). */
+  lemma LemmaConcatIsAssociative2<T>(a: seq<T>, b: seq<T>, c: seq<T>, d: seq<T>)
+    ensures a + b + c + d == a + (b + c + d)
+  {
+  }
+
+  /* The empty sequence is the neutral element of the concatenation operation. */
+  lemma Neutral(l: seq)
```

Comment:
nit: IMHO "Identity" is the more common term. Also consider "RightIdentity", or even "EmptySequenceIsRightIdentity".